client/my-sites/comments/comment-list/index.jsx

```diff
@@ -93,7 +103,11 @@ export class CommentList extends Component {
 
 	isCommentSelected = commentId => !! this.state.selectedComments[ commentId ];
 
-	isSelectedAll = () => this.props.comments.length === size( this.state.selectedComments );
+	isSelectedAll = () => this.getComments().length === size( this.state.selectedComments );
```

Comment:
Another leftover from the Faker days.
The Faker passed an ID-keyed object of comments instead of an array, and so I just reused the structure for `selectedComments`.

But because `persistedComments` must be merged with `props.comments` (which is an array), it just is cleaner to do all arrays.

I fully expected a comment about this weirdness, but again it's a change for another PR. 🙂